app/assets/javascripts/pretty-text/emoji.js.es6

```diff
@@ -43,43 +43,72 @@ export function performEmojiUnescape(string, opts) {
     return;
   }
 
-  return string.replace(unicodeRegexp, m => {
-    const isEmoticon = !!translations[m];
-    const isUnicodeEmoticon = !!replacements[m];
+  const inlineEmojiEnabled =
+    typeof Discourse !== "undefined"
+      ? Discourse.SiteSettings.enable_inline_emoji_translation
+      : !!opts.inlineEmojiEnabled;
+
+  let m;
+  while ((m = unicodeRegexp.exec(string)) !== null) {
+    const before = string.charAt(m.index - 1);
+    const isEmoticon = !!translations[m[0]];
+    const isUnicodeEmoticon = !!replacements[m[0]];
     let emojiVal;
     if (isEmoticon) {
-      emojiVal = translations[m];
+      emojiVal = translations[m[0]];
     } else if (isUnicodeEmoticon) {
-      emojiVal = replacements[m];
+      emojiVal = replacements[m[0]];
     } else {
-      emojiVal = m.slice(1, m.length - 1);
+      emojiVal = m[0].slice(1, m[0].length - 1);
     }
-    const hasEndingColon = m.lastIndexOf(":") === m.length - 1;
+    const hasEndingColon = m[0].lastIndexOf(":") === m[0].length - 1;
     const url = buildEmojiUrl(emojiVal, opts);
     const classes = isCustomEmoji(emojiVal, opts)
       ? "emoji emoji-custom"
       : "emoji";
-
-    return url && (isEmoticon || hasEndingColon || isUnicodeEmoticon)
-      ? `<img src='${url}' ${
-          opts.skipTitle ? "" : `title='${emojiVal}'`
-        } alt='${emojiVal}' class='${classes}'>`
-      : m;
-  });
+    const isAllowed =
```

Comment:
This is very hard to read/process/understand. Can it be split in several lines with either comments or (better) carefully named variables?